collect_app/src/main/java/org/odk/collect/android/widgets/UrlWidget.java

```diff
@@ -41,32 +42,43 @@
 @SuppressLint("ViewConstructor")
 public class UrlWidget extends QuestionWidget implements ButtonWidget {
```

Comment:
Does that still make sense to implement ButtonWidget? That was created in order to use WidgetViewUtils.createSimpleButton() - one method used in widgets for generating a button but if we are going to use xml files instead of creating those buttons programmatically we probably can get rid of it right?

Reply:
I do think having a separate method `onButtonClick()` is redundant. We can have all the functionality in `setOnClickListener()` instead of explicitly calling `onButtonClick()` from there. Should I remove it?

Reply:
I'm not talking about removing that method you can keep since you call it in https://github.com/getodk/collect/pull/3831/files#diff-a5c152dc7425469015c19b7a29b8da07R69
but implementing the interface and using `@Override` above the method doesn't make sense because this method is no longer called like before (using interface from WidgetViewUtils.createSimpleButton()) you do that directly from the same widget (the line I attached).

Reply:
Oh okay I understand. Thanks for explaining.